tools/reviewer/aa/AaScriptHelper.java

```diff
@@ -0,0 +1,173 @@
+
+  @Singleton
+  @Component(modules = {CommonModule.class, AaModule.class})
+  public interface ToolComponent {
+    AaScriptHelper getAaScriptHelper();
+  }
+
+  private String getMatches(ImmutableList<String> words, String word) {
+    return words
+        .stream()
+        .filter(x -> x.startsWith(word))
+        .sorted()
+        .collect(Collectors.joining("\n"));
+  }
+
+  private String getBashCompletions(String previousWord, String currentWord) throws IOException {
+    ImmutableList<String> commands = aaTool.getCommands();
+    // TODO: Get flags from commands. Perhaps AaTool can provide a get method for it.
+    ImmutableList<String> init_options =
+        ImmutableList.of("--base_path", "--startupos_repo", "--user");
+    ImmutableList<String> add_repo_options = ImmutableList.of("--url", "--name");
+    ImmutableList<String> diff_options =
+        ImmutableList.of("--reviewers", "--description", "--buglink");
+
+    if (previousWord.equals("aa")) {
+      return getMatches(commands, currentWord);
+    } else if (previousWord.equals("workspace") || previousWord.equals("aaw")) {
+      return getMatches(fileUtils.listSubfolders(workspacesPath), currentWord);
+    } else if (commands.contains(previousWord)) {
+      // Complete flags
+      switch (previousWord) {
+        case "init":
+          return getMatches(init_options, currentWord);
+        case "add_repo":
+          return getMatches(add_repo_options, currentWord);
+        case "diff":
+          return getMatches(diff_options, currentWord);
+      }
+    }
+    return "";
+  }
+
+  // bazelWorkspacePath is base/head/startup-os or base/ws/<some workspace>/startup-os
+  private void startServer(String bazelWorkspacePath) throws IOException {
+    if (!checkServer()) {
+      // TODO: Flags don't seem to work. Figure out and set --logToFile here and default to false.
+      String[] command = new String[] {"bazel-bin/tools/reviewer/local_server/local_server", "&"};
+      Runtime.getRuntime()
+          .exec(
+              command,
+              new String[] {"BUILD_WORKSPACE_DIRECTORY=" + bazelWorkspacePath},
+              new File(bazelWorkspacePath));
+      // TODO: Use AuthService.DEBUGGING_TOKEN_PATH. Consider moving it to a common location.
+      if (!fileUtils.fileExists("~/aa_token")) {
+        System.out.println("Visit https://web-login-startupos.firebaseapp.com to log in");
+      }
+      try {
+        // Wait until server is up
+        while (!checkServer()) {
+          Thread.sleep(50);
+        }
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  // bazelWorkspacePath is base/head/startup-os or base/ws/<some workspace>/startup-os
+  private boolean checkServer() {
+    try {
+      return "OK".equals(httpUtils.get("http://localhost:7000/health"));
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  private void check(boolean predicate, String message) throws IllegalArgumentException {
```

Comment:
@oferb strictly speaking, I think it's not `predicate`, it's the result of predicate evaluation. Citing [source](https://en.wikipedia.org/wiki/Predicate_(mathematical_logic))
>Predicate is commonly understood to be a Boolean-valued function P: X→ {true, false}, called the predicate on X. 

But I'm ok with it because function is small and concise 🙂

Reply:
Good point. Changed to `predicateResult` :)